### PR TITLE
CSHARP-3331: Matrix Testing - MongoDB 4.2-era Driver Support

### PR DESCRIPTION
**Pull request by MikalaiMazurenka**
JIRA: https://jira.mongodb.org/browse/CSHARP-3331
EPIC: https://jira.mongodb.org/browse/CSHARP-3328
Evergreen: https://evergreen.mongodb.com/version/6059145d2fbabe7ca00d2909

### CODE REVIEW

File: COMPATIBILITY.MD [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+5. `CreateIndexesOperation` result field `expireAfterSeconds` value is rounded since 4.9.0.
+See respective ticket: [CSHARP-3294: Fix CreateIndexesOperationTests tests on latests](https://jira.mongodb.org/browse/CSHARP-3294)
+6. 🦖 `CreateIndex` with `WildcardProjection` reports result `_id` value as boolean (`{ _id : true }`) instead of int (`{ _id : 1 }`) starting from 4.5.0. That's fine, since it is legacy API.
+See respective ticket: [CSHARP-3078: Fix test CreateIndex_with_wildcardProjection_should_create_expected_index on latest](https://jira.mongodb.org/browse/CSHARP-3078)
+7. 🦖 `ReIndex` command fails on replica set configuration, but works on standalone as it used to. This operation is part of legacy API.
+See respective change in 2.11.0 driver: [CSHARP-3058: Make 'reIndex' a standalone-only command](https://jira.mongodb.org/browse/CSHARP-3058)
+8. `trunkatedArrays` field is not handled in `ChangeStreamDocument.UpdateDescription` property deserialization, causing the access to the property throw and exception `System.FormatException : Invalid field name: "truncatedArrays".` since server version 4.9.0. Other properties work as intended.
```
Comment (JamesKovacs) [inline COMPATIBILITY.MD:20]: `trunkatedArrays` => `truncatedArrays`

**Review by JamesKovacs — CHANGES_REQUESTED** [review]
LGTM but with minor spelling correction.

**Comment by MikalaiMazurenka** [comment]
Ready for review!

**Review by JamesKovacs — APPROVED** [review]
LGTM

**Pull request merged**